### PR TITLE
DOC: add warning section in indexing docs. 

### LINKED ISSUE #12947: Confusing behavior with (multi-)assignment and _LocIndexer/_IXIndexer
```python
df = pandas.DataFrame([[1,2,3,4,5]], columns=["A", "B", "C", "D", "E"])
# Suppose you want to set df.B to df.C when df.A ==1
# then the following both work:
df.loc[df.A== 1, "B"] = df.loc[df.A == 1, "C"]
df.ix[df.A == 1, "B"] = df.ix[df.A == 1, "C"]
# you can even mix and match them with ix on one side and loc on the other

# but maybe you have two or more columns you want to set... Its natural to think that:
df.loc[df.A== 1, ["B", "C"]] = df.loc[df.A == 1, ["D", "E"]]
df.ix[df.A == 1, ["B", "C"]] = df.ix[df.A == 1, ["D", "E"]]
# but they actually just NaN out df.B and df.C (it isn't an issue of a silent copy losing updates)
# in fact the application of NaN even happens if you have singletons
df.loc[df.A== 1, ["D"]] = df.loc[df.A == 1, ["E"]]

# presumably because
type(df.ix[df.A == 1, "B"])
# is pandas.core.series.Series but 
type(df.ix[df.A == 1, ["B"]])
# is pandas.core.frame.DataFrame
# but when printed they look really similar... 
# 0    3
# Name: B, dtype: int64
# versus
#    B
# 0  3
# so it is easy to get confused

# If this can't be made to work in the natural fashion it would be a lot nicer if it could just throw an error
# the same way the following does:
df.ix[df.A == 1, ["B", "C"]] = df[df.A == 1]["D", "E"] 
```

INSTALLED VERSIONS
------------------
commit: None
python: 3.5.1.final.0
python-bits: 64
OS: Linux
OS-release: 2.6.32-431.29.2.el6.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.18.0
nose: 1.3.7
pip: 8.1.1
setuptools: 20.4
Cython: None
numpy: 1.11.0
scipy: None
statsmodels: None
xarray: None
IPython: 4.1.2
sphinx: None
patsy: None
dateutil: 2.5.2
pytz: 2016.3
blosc: None
bottleneck: None
tables: None
numexpr: None
matplotlib: None
openpyxl: 2.3.4
xlrd: None
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.12
pymysql: None
psycopg2: None
jinja2: None
boto: None

### PR DESCRIPTION
**Pull request by zhangxiangnick**
Add warning section in indexing docs. Close #12947.

### CODE REVIEW

File: doc/source/indexing.rst [hunk 1]
```diff
@@ -191,6 +191,17 @@ raised. Multiple columns can also be set in this manner:
...
+.. warning::
+
+   Pandas aligns index for assigning ``DataFrame`` from ``.loc``, ``.iloc`` and ``.ix``. 
+   For example, in the expression
+
+   .. code-block:: python
+
+      df.loc[:,['B', 'A']] = df.loc[:,['A', 'B']]
+
+   ``df`` is not modified. The alignment is ignored in column assignment.
```
Comment (jorisvandenbossche) [inline doc/source/indexing.rst]: It would maybe be good to also give an illustration of the last sentence

**Comment by jreback** [comment]
yeah this needs a bit more documentation of *why* this doesn't work and then what you should do instead.

**Comment by zhangxiangnick** [comment]
Add why the DataFrame assignment may not work and then the correct to do it. Anything else I can do?

File: doc/source/indexing.rst [hunk 2]
```diff
@@ -191,6 +191,29 @@ raised. Multiple columns can also be set in this manner:
 You may find this useful for applying a transform (in-place) to a subset of the
 columns.
 
+.. warning::
+
+   pandas aligns index for setting a ``DataFrame`` from ``.loc``, ``.iloc`` and ``.ix``.
```
Comment (jreback) [inline doc/source/indexing.rst]: aligns for when setting (not specific to ``DataFrame``)

File: doc/source/indexing.rst [hunk 3]
```diff
@@ -191,6 +191,29 @@ raised. Multiple columns can also be set in this manner:
...
 
+.. warning::
+
+   pandas aligns index for setting a ``DataFrame`` from ``.loc``, ``.iloc`` and ``.ix``. 
+   For example, the expression
+
+   .. code-block:: python
+
+      df.loc[:,['B', 'A']] = df.loc[:,['A', 'B']]
+
```
Comment (jreback) [inline doc/source/indexing.rst]: I would show the rhs here (as well)

Comment (zhangxiangnick) [inline doc/source/indexing.rst]: What do you mean by "show here"?

Comment (jreback) [inline doc/source/indexing.rst]: I would show ``df.loc[:, ['A', 'B']`` (e.g. what you are assigning)

**Pull request closed without merging**